provisioning/provisioning-device-client/src/main/java/com/microsoft/azure/sdk/iot/provisioning/device/internal/task/ProvisioningTask.java

```diff
@@ -343,7 +343,7 @@ public Object call() throws Exception
             this.executeStateMachineForStatus(registrationOperationStatusParser);
             this.close();
         }
-        catch (ExecutionException | TimeoutException | ProvisioningDeviceClientException | SecurityProviderException e)
+        catch (ExecutionException | TimeoutException | ProvisioningDeviceClientException | SecurityProviderException | InterruptedException e)
```

Comment:
This catch block cleans up all local resources including the RegisterTask thread. The problem here was that this thread would be interrupted and then exit without ever closing the threads it started.

Now we end all threads this thread created before ending this thread